src/test/java/book_store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,233 @@
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto[].class);
+        Assertions.assertEquals(3, actual.length);
+        for (BookDto book : expected) {
+            Assertions.assertTrue(Arrays.stream(actual)
+                    .anyMatch(a -> a.getId().equals(book.getId())
+                            && a.getTitle().equals(book.getTitle())
+                            && a.getAuthor().equals(book.getAuthor())
+                            && a.getPrice().equals(book.getPrice())
+                            && a.getIsbn().equals(book.getIsbn())));
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get book by ID")
+    void getBookById_ReturnsBookById_ExpectedSuccess() throws Exception {
+        BookDto expectedBook = new BookDto()
+                .setId(1L)
+                .setTitle("Kobzar")
+                .setAuthor("Shevchenko")
+                .setPrice(BigDecimal.valueOf(10.99))
+                .setIsbn("9460303332081");
+
+        MvcResult result = mockMvc.perform(
+                        get("/books/{id}", 1L)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actualBook = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto.class);
+
+        Assertions.assertEquals(expectedBook.getId(), actualBook.getId());
+        Assertions.assertEquals(expectedBook.getTitle(), actualBook.getTitle());
+        Assertions.assertEquals(expectedBook.getAuthor(), actualBook.getAuthor());
+        Assertions.assertEquals(expectedBook.getPrice(), actualBook.getPrice());
+        Assertions.assertEquals(expectedBook.getIsbn(), actualBook.getIsbn());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(
+            scripts = "classpath:database/books/delete-John_Doe-book.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Create a new book")
+    void createBook_CreatesNewBook_ExpectedSuccess() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setAuthor("John Doe")
+                .setTitle("Book Title")
+                .setDescription("Book Description")
+                .setIsbn("9460306332081")
+                .setPrice(BigDecimal.valueOf(10.99));
+
+        BookDto expected = new BookDto()
+                .setAuthor(requestDto.getAuthor())
+                .setTitle(requestDto.getTitle())
+                .setDescription(requestDto.getDescription())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice());
```

Comment:
what is sense of having 2 identical dtos? you may omit expected and use requestDto as reference